Mathlib/GroupTheory/ArchimedeanDensely.lean

```diff
@@ -210,3 +210,36 @@ lemma LinearOrderedCommGroup.discrete_or_denselyOrdered :
   refine (LinearOrderedAddCommGroup.discrete_or_denselyOrdered (Additive G)).imp ?_ id
   rintro ⟨f, hf⟩
   exact ⟨AddEquiv.toMultiplicative' f, hf⟩
+
+/-- Any nontrivial (has other than 0 and 1) linearly ordered mul-archimedean group with zero is
+either isomorphic (and order-isomorphic) to `ℤₘ₀`, or is densely ordered. -/
+lemma LinearOrderedCommGroupWithZero.discrete_or_denselyOrdered (G : Type*)
+    [LinearOrderedCommGroupWithZero G] [Nontrivial Gˣ] [MulArchimedean G] :
+    Nonempty (G ≃*o WithZero (Multiplicative ℤ)) ∨ DenselyOrdered G := by
```

Comment:
Can you not use the notation here?
```suggestion
    Nonempty (G ≃*o ℤₘ₀) ∨ DenselyOrdered G := by
```

Reply:
`ℤₘ₀` is defined all the way later in a `Valuation` folder file

Reply:
Uh, maybe that should be fixed? It doesn't need any more than `Algebra.Group.Int`

Reply:
Will make separate PR

Reply:
#16873